src/components/side-panel/data-selection/dimension-selection/DimensionSelection.tsx

```diff
@@ -76,18 +77,27 @@ const DimensionSelection = ({ availableDimensions }: Props): JSX.Element => {
           name: defaultSelectedDimension,
           displayName: titleCase(defaultSelectedDimension),
           color: colors[selectedDimensions.length],
+          intervalColor: colors[selectedDimensions.length],
           dashStyle: "none",
+          intervalType: "---",
```

Comment:
I've changed intervalType choice to a const called INTERVAL_STYLES, with a plan to also change intervalType to intervalStyle as this makes more sense to read.

I'm a little unfamiliar with TypeScript syntax so at a couple points had to use `INTERVAL_STYLES[0].toString()` to get it to assign the value rather than the type, I'll need to follow up with this to ensure this is the best way to go about it.

Reply:
I guess there are two separate things here.

As per usual, take it with a pinch of "Mike doesn't understand frontend yet" salt.

## Constants for 1-n choices

here's a nice trick for this using enums. I _think_ this is the relevant example for typescript: https://www.typescriptlang.org/docs/handbook/enums.html.

something like

```
enum IntervalStyle {
  None "---"
  Intervals "Intervals"
  ErrorBars "Error Bars"
 }
```

to create a static referenceable set of constant values.

so you can reference it as (for example) `if (series.IntervalStyle == IntervalStyles.ErrorBars) {` . That gets us away from directly referencing strings.

you can probably apply that pattern anywhere you find yourself defining the same string over and over again (so you change it in one place, and  can never get caught out by typos, its also super useful for sustainability as someone coming to the code later can just look up the viable values for any given field).

## Naming

I think I may have made this messier for you with a slightly vague comment by all means put it back to what it was, but what I was trying to say is if we're giving the user choices on how to display intervals then one of those options being "intervals" doesn't quite make sense

"What type of socks are you wearing Mike?"
"Socks!"

"What's your favourite tv show Mike?"
"Tv show!"

see what I mean? 😄 

It gets tricky on the frontend because if that makes sense to the user then "intervals" it is I guess 🤷 , the former point about getting away from comparing to strings is way more important.

Reply:
Thanks for the tip Mike, switched it over to enum!

Ah yes, I get what you mean. I think the whole section could potentially need a rename away from confidence intervals.
What would be a more suitable name? Uncertainty Indicators? Error Bars and Confidence Intervals?
I'll raise this in standup.